src/gfx/renderJob/collect/EntityCollect.ts

```diff
@@ -278,9 +278,11 @@ export class EntityCollect {
                 //resume unchange status
                 renderNode.isRenderOrderChange = false;
             }
-            renderList.sort((a: RenderNode, b: RenderNode) => {
-                return a['__renderOrder'] > b['__renderOrder'] ? 1 : -1;
-            });
+
+            const sortFn: (a: RenderNode, b: RenderNode) => number = Engine3D.setting.render.useReversedDepth
+                ? (a, b) => (a['__renderOrder'] - b['__renderOrder'] > 0 ? -1 : 1)
+                : (a, b) => (a['__renderOrder'] - b['__renderOrder'] > 0 ? 1 : -1);
```

Comment:
You can simplify the sort functions, remove the ternary, and also cover the missing case of equals (0) just by using subtraction.

```suggestion
                ? (a, b) => b['__renderOrder'] - a['__renderOrder']
                : (a, b) => a['__renderOrder'] - b['__renderOrder'];
```

Reply:
I gotcha. Maybe `__renderOrder` can never be equals between two RenderNode's, but if it does, then the sort will be unstable. Plus, branching for every comparison seems unoptimized, when it can be just handled by subtraction.

Reply:
yeah, i agree with all your points. i'll change it!